rtCommon/bidsCommon.py

```diff
@@ -0,0 +1,211 @@
+
+
+def isNiftiPath(path: str) -> bool:
+    """
+    Returns true if the provided path points to an uncompressed NIfTI file,
+    false otherwise.
+    """
+    _, ext = os.path.splitext(path)
+    return ext == '.nii'
+
+
+def isJsonPath(path: str) -> bool:
+    """
+    Returns true if the provided path points to an uncompressed NIfTI file,
+    false otherwise.
+    """
+    _, ext = os.path.splitext(path)
+    return ext == '.json'
+
+
+@functools.lru_cache(maxsize=128)
+def makeDicomFieldBidsCompatible(field: str) -> str:
+    """
+    Remove characters to make field a BIDS-compatible
+    (CamelCase alphanumeric) metadata field.
+
+    NOTE: Keys like 'Frame of Reference UID' become 'FrameofReferenceUID', which
+    might be different than the expected behavior
+    """
+    return re.compile('[^a-zA-z]').sub("", field)
+
+
+def adjustTimeUnits(imageMetadata: dict) -> None:
+    """
+    Validates and converts in-place the units of various time-based metadata,
+    which is stored in seconds in BIDS, but often provided using milliseconds in
+    DICOM.
+    """
+    fieldToMaxValue = {"RepetitionTime": 100, "EchoTime": 1}
+    for field, maxValue in fieldToMaxValue.items():
+        value = imageMetadata.get(field, None)
+        if value is None:
+            continue
+        else:
+            value = int(value)
+
+        if value <= maxValue:
+            continue
+        elif value / 1000.0 <= maxValue:
+            logger.info(f"{field} has value {value} > {maxValue}. Assuming "
+                        f"value is in milliseconds, converting to seconds.")
+            value = value / 1000.0
+            imageMetadata[field] = value
+        else:
+            raise ValidationError(f"{field}'s max value is {maxValue}; {value} "
+                                  f"> {maxValue} even if interpreted as "
+                                  f"milliseconds.")
+
+
+def metadataFromProtocolName(protocolName: str) -> dict:
+    """
+    Extracts BIDS label-value combinations from a DICOM protocol name, if
+    any are present.
+
+    Returns:
+        A dictionary containing any valid label-value combinations found.
+    """
+    if not protocolName:
+        return {}
+
+    prefix = "(?:(?<=_)|(?<=^))"  # match beginning of string or underscore
+    suffix = "(?:(?=_)|(?=$))"  # match end of string or underscore
+    fieldPat = "(?:{field}-)(.+?)"  # TODO(spolcyn): Document this regex
+    patternTemplate = prefix + fieldPat + suffix
+
+    foundEntities = {}
+    for entityName, entityValueDict in loadBidsEntities().items():
+        entity = entityValueDict[BidsEntityKeys.ENTITY.value]
+        entitySearchPattern = patternTemplate.format(field=entity)
+        result = re.search(entitySearchPattern, protocolName)
+
+        if result is not None and len(result.groups()) == 1:
+            foundEntities[entityName] = result.group(1)
+
+    return foundEntities
+
+
+def getMetadata(dicomImg: pydicom.dataset.Dataset) -> (dict, dict):
```

Comment:
I'd suggest including Dicom in the function name, such as getDicomMetadata. Perhaps for other functions too if appropriate.